### ISSUE TITLE
AWS Migration [Data Register] [DUE: October 2025]

### ISSUE THREAD

**Issue by LindsayMacfarlane** [bcgov/parks-data-register#554]
#### Description:
We must migrate from AWS ASEA to LZA by October 31, 2025
Launch day: June 24, 2025.
AWS support countdown: September - October, 2025

Please review info in https://github.com/bcgov/parks-reso-api/issues/406 and in the migration Epic for more deets to help with refinement.

#### Acceptance Criteria:
Given we are a product team using AWS ASEA
When it is time to migrate
Then we 'move' over to the new LZA AWS instance.

#### Development Checklist:

- [x] Create new project sets for LZA in the registry
- [x] Obtain expense authority approval (eMOU process) (note: depending on value, this will either be Lindsay or Rumon)
- [x] Provision project sets
- [x] Replicate workloads in Dev/Test environments
- [x] Validate Dev/Test deployments
- [x] Create production workloads in LZA
- [x] Plan and execute your production switch
- [x] Validate production workloads
- [ ] Decommission ASEA project sets

#### Dependencies

- Blocked by https://github.com/bcgov/bcparks/issues/62
- Blocking

**Relevant documentation as reference**
- [LZA kickoff meeting presentation](https://bcgov-my.sharepoint.com/:p:/r/personal/khurrum_husain_gov_bc_ca/Documents/Microsoft%20Teams%20Chat%20Files/LZA%20kick-off%20draft%20slides.pptx?d=wd7f6c556159a4f288e0f8cb25dade935&csf=1&web=1&e=TPeQCB)

**Definition of Ready**

- [x] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [x] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by LindsayMacfarlane** [comment]
Followed up with Team Osprey on April 22, 2025 - From our understanding, our 'projects' need to be created by the platform team and until then we are blocked on moving forward. Assumption is that it will happen sometime in FY 2025-26. Added the blocked label to this ticket.

**Comment by marklise** [comment]
# LZA Environment URLs

This document contains the CloudFront and API Gateway URLs for the LZA-managed Parks Data Register environments.

## LZA-DEV Environment

### Admin (Frontend)
- **CloudFront URL**: https://d90ldtb7xzcdw.cloudfront.net
- **CloudFront Distribution ID**: E3G59T8MH280VJ

### API (Backend)
- **API Gateway URL**: https://gw4ohkljkk.execute-api.ca-central-1.amazonaws.com/lza-dev/
- **Config Endpoint**: https://gw4ohkljkk.execute-api.ca-central-1.amazonaws.com/lza-dev/config/

---

## LZA-TEST Environment

### Admin (Frontend)
- **CloudFront URL**: https://d3d31c234iruwb.cloudfront.net
- **CloudFront Distribution ID**: EXPE3MB1JDSYE

### API (Backend)
- **API Gateway URL**: https://e8qk6n9teg.execute-api.ca-central-1.amazonaws.com/lza-test/
- **Config Endpoint**: https://e8qk6n9teg.execute-api.ca-central-1.amazonaws.com/lza-test/config/

---

## Authentication

### Keycloak SSO
- **lza-dev**: https://dev.loginproxy.gov.bc.ca/auth
- **lza-test**: https://test.loginproxy.gov.bc.ca/auth
- **Realm**: bcparks-service-transformation
- **Client**: data-register

### API Keys
- **lza-dev**: jQD4xxcHit7S9YaZIKsor7cJsvPpfMs3hDsQFVuh (Usage Plan: ajisvq)
- **lza-test**: G73PtZ9l6t53rJYNperGb7MngNd88HpS8qYnaNiq (Usage Plan: 21mwb2)

---

## Notes

- All environments are deployed in the **ca-central-1** (Canada Central) AWS region
- CloudFormation stacks:
  - `pdr-api-lza-dev` (Status: UPDATE_COMPLETE)
  - `pdr-admin-lza-dev` (Status: UPDATE_COMPLETE)
  - `pdr-api-lza-test` (Status: UPDATE_COMPLETE)
  - `pdr-admin-lza-test` (Status: UPDATE_COMPLETE)
- AWS Account: 281145281748 (BCGOV LZA Admin)
- All stack names now follow the pattern: `pdr-{component}-lza-{environment}`
- Frontend uses direct API Gateway access (bypassing CloudFront for API calls)
- DynamoDB tables: NameRegister-lza-dev, NameRegister-lza-test, Audit-lza-dev, Audit-lza-test
- **lza-prod environment still needs to be deployed**

---

**Last Updated**: October 31, 2025

**Comment by manuji** [comment]
Tested on TEST environment: **Passed**
* Verified that the url to new TEST admin site is working: https://d3d31c234iruwb.cloudfront.net/
* Verified that the data displayed accurately
* Verified that the minor change, major change and Repeal actions are working as expected. 
* Verified that the search on both Protected area and change log work as expected. 
* This ticket is good to go to PO review and PROD. 
FYI @marklise  & @LindsayMacfarlane

**Comment by marklise** [comment]
## ✅ LZA Production Environment Deployed

The new LZA production environment has been successfully deployed to AWS account `183300739488`.

### LZA-PROD Environment

**Admin (Frontend):**
- CloudFront URL: https://d27wzcea98giei.cloudfront.net
- CloudFront Distribution ID: E9FPYMCIQKCIZ

**API (Backend):**
- API Gateway URL: https://h8fw0jn7bc.execute-api.ca-central-1.amazonaws.com/lza-prod/
- Config Endpoint: https://h8fw0jn7bc.execute-api.ca-central-1.amazonaws.com/lza-prod/config/

**Deployment:**
- GitHub Actions workflows trigger on git tags (v*)

**Comment by LindsayMacfarlane** [comment]
Another huge milestone! Congrats @marklise !

**Comment by manuji** [comment]
Smoke tested on PROD environment and everything looks good. Did not edit any data as this is the PROD environment. Thanks @marklise .

**Comment by LindsayMacfarlane** [comment]
Sent email to Glenna and Rob regarding migration completion and to let us know if they experience any issues. Since in PROD, moving this ticket to done.

**Comment by marklise** [comment]
This is not done yet - there are things left to do to notify and allow existing connections to the DR to be wired up.

dev-data, test-data, and data.bcparks.ca are now live.

API Keys have been migrated, but users of them need to be notified and have their pointers tested.

Decommissioning also has not been completed.

**Comment by marklise** [comment]
New issue found - we cannot decommission the old accounts until the github.io page located at bcgov.github.io/bcparks is migrated to a new account. This is being piggybacked into the DR account.

**Comment by LindsayMacfarlane** [comment]
Moving this back to do out of blocked column. The github.io pages have now been moved, so we should be able to wrap up this work :)

**Comment by marklise** [comment]
Marked for project set deletion.

**Comment by LindsayMacfarlane** [comment]
Woohoo @marklise ! Congratulations - huge accomplishment!
